grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/OntologicalAtom.java

```diff
@@ -58,4 +55,24 @@ public boolean isSelectable() {
     public Set<String> validateAsRuleHead(Rule rule) {
         return Sets.newHashSet(ErrorMessage.VALIDATION_RULE_ILLEGAL_ATOMIC_IN_HEAD.getMessage(rule.getThen(), rule.getLabel()));
     }
+
+    @Override
+    public Set<TypeAtom> unify(Unifier u){
+        Collection<Var> vars = u.get(getVarName());
+        return vars.isEmpty()?
+                Collections.singleton(this) :
+                vars.stream().map(v -> AtomicFactory.createOntologicalAtom(this.getClass(), v, getPredicateVariable(), getTypePredicate(), this.getParentQuery())).collect(Collectors.toSet());
```

Comment:
Maybe this will work instead of `createOntologicalAtom`...

Add an abstract method `OntologicalAtom#createSelf` (or a better name):

```
abstract OntologicalAtom createSelf(Var var, Var predicateVar, @Nullable IdPredicate predicate, ReasonerQuery parent);
```

Then implement it for each atom (four times?):

```
class PlaysAtom {
    ...
    OntologicalAtom createSelf(Var var, Var predicateVar, @Nullable IdPredicate predicate, ReasonerQuery parent) {
        returns PlaysAtom.create(var, predicateVar, predicate, parent);
    }
}
```
I guess it's a bit more code, but it might save us from the scary `getClass()` stuff 😛